compass/landice/tests/ismip6_run/ismip6_ais_proj2300/set_up_experiment.py

```diff
@@ -268,5 +310,95 @@ def run(self):
         """
         Run this step of the test case
         """
+
+        config = self.config
+        logger = self.logger
+        section = config['ismip6_run_ais_2300']
+        sea_level_model = section.getboolean('sea_level_model')
+        if sea_level_model:
+            self._build_mapping_files(config, logger)
```

Comment:
I agree it makes sense to build the mapping files in the run method, because it will typically be more expensive than we want to do on a login node.  A problem with putting it here, though, is the mapping files would get regenerated if `compass run` was used on a restart, which is what happens when the generated job script is used.  This is kind of a symptom of using compass outside its intended usage.  One solution could be to have `_build_mapping_files` not do anything if the mapping file already exists (as would be the case on a restart).  That seems a little fragile, but less clunky than having to tell compass run if it's a restart or not.  @xylar, do you have any advice for this situation?

Reply:
One option would be to move `_build_mapping_files` to a separate step.  This would have the advantage of only needing to be run once, instead of right now it will be rerun for every experiment.  The experiment steps could contain a symlink to the mapping files in the build_mapping step.  A user would have to know to execute `compass run` for the build_mapping step before starting any of the runs (this would have to be documented in the users guide).  If a user forgot that step, MALI would die when the nonexistent mapping file tried to be read, which would alert the user of the problem.  I think I prefer this solution, but I'm curious to hear if @hollyhan or @xylar can think of any objections.  (@hollyhan , if we agree to go this route, I'm willing to implement this change so you don't have to deal with that)

Reply:
Thanks for making the note, @matthewhoffman!  I'm quite neutral about making the further changes. In one hand, I have no problem with re-running the build_mapping routine in the case of restart because it's relatively a cheap and short process, and it's likely that one would run the simulations by directly executing `srun ./landice_model` rather than `compass run` as that was the initial intention of this PR (following your previous PR that sets up standalone MALI simulations), which would avoid re-creating the mapping files. But I'm also happy if you want to make changes :) .

Reply:
Again, sorry for losing track of this comment.

I'm more inclined to @matthewhoffman's solution of having a separate step for making the mapping files.  Yes, it's cheap to run in this case and it's a little annoying for a user to have to run an extra step.  But it's more consistent with the philosophy of how Compass is supposed to work.

I also don't think we want users to get in the habit of circumventing `compass run`.  It's okay for a knowledgeable user to do that but it shouldn't be required by default because it requires more manual intervention (or at least it normally does -- I'm not sure I'm up on the details of this test group anymore).

Reply:
Okay. I am convinced that we take the route that Matt suggested. @matthewhoffman - would you mind implementing the solution? My bandwidth is super tight for the next few weeks, it would be really helpful if you could take care of this part. :)

Reply:
@hollyhan , yes, I can do that part!